### ISSUE TITLE
Git blame not working on Sublime Text 3

### ISSUE THREAD

**Issue by maulikfojdar** [frou/st3-gitblame#25]
I get an error while using git blame in sublime.

`Git blame: git error 128:
fatal: no such ref: HEAD`

**Comment by psykzz** [comment]
Could you give some more details?
What happens if you run something like `git status` in the repo root?

**Comment by maulikfojdar** [comment]
git status gives me this:

```
On branch master
Your branch is up-to-date with 'origin/master'.
nothing to commit, working directory clean
```

**Comment by psykzz** [comment]
Looking into this a little it looks like it may be related to this issue. https://stackoverflow.com/q/4848607/211081

Could you try the following and let me know if it resolves your issue?
`echo ref: refs/heads/master > .git/HEAD`

**Comment by psykzz** [comment]
Any luck @maulikfojdar?

**Comment by maulikfojdar** [comment]
No that is not the issue. I have used that thread a long time back. My git blame is not working since 2-3 months now. There is no HEAD ref issue here. And my colleagues are also facing the same issue since a long time. 

So I did change my configuration some time back. I code on sublime in OSX but the git repo is in a headless VM which I have mounted on my OSX. I wonder if that is the source of this issue. Well I have also tried git blame on my ubuntu VM sublime where git repo was cloned. 

Another observation is that, there is a Git package for sublime which I have tried to use when your git blame stopped working. That also have blame function which results in same error. But when I use git blame on terminal, it works just fine. It just does not work on my sublime.

**Comment by psykzz** [comment]
You may have some `.bash_profile` scripts help it to work locally which are not being picked up by Sublime?

Honestly, I'm not sure of the solution.

**Comment by yamoinza** [comment]
I just figured out why this started happening to me in one of my workspaces...
I found an "extraneous" .git folder in the hierarchy to the current file.
Note that in my case at least, the command that is shown as failing does not include the complete path, just the filename itself. If I did a git blame at the root of the workspace with the path to the file then blame worked.
but if I cd-ed to the dir containing the file and did a blame using filename only then I got the same error.
The "bad" .git folder had a reference to master but my workspace was on a dev branch so this was real confusing.
I don't know the source of the extraneous .git folder but regardless if you come to this page that is something to look for as in:
find . -type d -name .git
./.git
./client/.git <----- this one was unexpected
